service/src/main/java/org/cbioportal/service/impl/GenePanelServiceImpl.java

```diff
@@ -172,37 +178,106 @@ public GenePanel getGenePanel(String genePanelId) throws GenePanelNotFoundExcept
         for(int i = 0; i < sampleIds.size(); i++) {
             String sampleId = sampleIds.get(i);
             String molecularProfileId = molecularProfileIds.get(i);
-        
-            GenePanelData resultGenePanelData = new GenePanelData();
             MolecularProfile molecularProfile = molecularProfileMap.get(molecularProfileId);
             String studyId = molecularProfile.getCancerStudyIdentifier();
-            resultGenePanelData.setStudyId(studyId);
-            Optional<GenePanelData> genePanelData =
-                Optional.ofNullable((GenePanelData)genePanelDataMap.get(molecularProfileId, sampleId));
-            if (genePanelData.isPresent()) {
-                resultGenePanelData.setGenePanelId(genePanelData.get().getGenePanelId());
-                resultGenePanelData.setProfiled(true);
-            } else {
-                List<SampleList> sampleLists = sampleListMap.get(studyId);
-                if (molecularProfile.getMolecularAlterationType().equals(MolecularProfile.MolecularAlterationType.MUTATION_EXTENDED) && 
-                    (sampleLists == null || (sampleLists != null && sampleLists.get(0).getSampleIds().contains(sampleId)))) {
+            
+            if (samples.containsKey(sampleId, studyId)) {
+                GenePanelData resultGenePanelData = new GenePanelData();
+                resultGenePanelData.setStudyId(studyId);
+                Optional<GenePanelData> genePanelData =
+                    Optional.ofNullable((GenePanelData)genePanelDataMap.get(molecularProfileId, sampleId));
+                if (genePanelData.isPresent()) {
+                    resultGenePanelData.setGenePanelId(genePanelData.get().getGenePanelId());
                     resultGenePanelData.setProfiled(true);
                 } else {
-                    resultGenePanelData.setProfiled(false);
+                    List<SampleList> sampleLists = sampleListMap.get(studyId);
+                    if (molecularProfile.getMolecularAlterationType().equals(MolecularProfile.MolecularAlterationType.MUTATION_EXTENDED) && 
+                        (sampleLists == null || (sampleLists != null && sampleLists.get(0).getSampleIds().contains(sampleId)))) {
+                        resultGenePanelData.setProfiled(true);
+                    } else {
+                        resultGenePanelData.setProfiled(false);
+                    }
```

Comment:
You can get rid of this whole if/else and just stick the predicate in `setProfiled`